### PR TITLE
Fetch summary only when refSeq number advances.

### LINKED ISSUE #7560: Fetch summary only when reference sequence number advances
Currently, refreshLatestSummary() downloads the summary and checks whether the reference sequence number advances (https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts#L270). However, the summarizer already knows the reference sequence number from the last summary op (https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/container-runtime/src/summarizer.ts#L275).

Rather than downloading the summary, we can just check the condition and only download summary when the refSeq actually advances. This will also help service summaries since summarizer will stop pulling older summaries unnecessarily just to perform the check.

### PR DESCRIPTION
**Pull request by tanviraumi**
Fixes #7560

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -2239,6 +2241,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
         const result = await this.summarizerNode.refreshLatestSummary(
             undefined,
+            snapshotRefSeq,
```
Comment (hedasilv) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2244]: Any reason to call it `snapshotRefSeq` in here instead of `summaryRefSeq`?

Comment (tanviraumi) [reply packages/runtime/container-runtime/src/containerRuntime.ts:2244]: They are used interchangeably in runtime. Did not want to tweak existing variable :)

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 2]
```diff
@@ -2194,12 +2194,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     /** Implementation of ISummarizerInternalsProvider.refreshLatestSummaryAck */
     public async refreshLatestSummaryAck(
         proposalHandle: string | undefined,
+        summaryRefSeq: number | undefined,
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:2197]: This cannot be undefined. We don't process summary ack without an op.

File: packages/runtime/container-runtime/src/summarizerTypes.ts [hunk 3]
```diff
@@ -56,6 +56,7 @@ export interface ISummarizerInternalsProvider {
     /** Callback whenever a new SummaryAck is received, to update internal tracking state */
     refreshLatestSummaryAck(
         proposalHandle: string,
+        summaryRefSeq: number,
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/summarizerTypes.ts:59]: nit: Consider adding this after the `ackHandle` param so the two handles are together.

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts [hunk 4]
```diff
@@ -263,6 +264,11 @@ export class SummarizerNode implements IRootSummarizerNode {
             }
         }
 
+        // If we have seen a summary same or later as the current one, ignore it.
+        if (summaryRefSeq !== undefined && this.referenceSequenceNumber >= summaryRefSeq) {
+            return { latestSummaryUpdated: false };
+        }
+
         const snapshotTree = await getSnapshot();
         const referenceSequenceNumber = await seqFromTree(snapshotTree, readAndParseBlob);
```
Comment (agarwal-navin) [inline packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:273]: This is not needed anymore. `summaryRefSeq` should be used instead.

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts [hunk 5]
```diff
@@ -250,6 +250,7 @@ export class SummarizerNode implements IRootSummarizerNode {
      */
     public async refreshLatestSummary(
         proposalHandle: string | undefined,
+        summaryRefSeq: number | undefined,
```
Comment (agarwal-navin) [inline packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:253]: This cannot be undefined.

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts (hunk 4 above)
Comment (agarwal-navin) [reply packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:273]: The check below should also be removed.

File: packages/runtime/runtime-utils/src/test/summarizerNode.spec.ts [hunk 6]
```diff
@@ -330,6 +330,7 @@ describe("Runtime", () => {
                 it("Should refresh from tree when no proposal handle provided", async () => {
                     createRoot();
                     const result = await rootNode.refreshLatestSummary(
+                        undefined,
```
Comment (agarwal-navin) [inline packages/runtime/runtime-utils/src/test/summarizerNode.spec.ts:333]: You can use the sequenceNumber of the "attributes" blob defined in the test:
const blobs = {
                protocolAttributes: { sequenceNumber: 123 },
            } as const;

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts [hunk 7]
```diff
@@ -263,14 +265,14 @@ export class SummarizerNode implements IRootSummarizerNode {
...
+        // If we have seen a summary same or later as the current one, ignore it.
+        if (this.referenceSequenceNumber >= summaryRefSeq) {
             return { latestSummaryUpdated: false };
         }
 
+        const snapshotTree = await getSnapshot();
+        const referenceSequenceNumber = await seqFromTree(snapshotTree, readAndParseBlob);
+
         await this.refreshLatestSummaryFromSnapshot(
             referenceSequenceNumber,
```
Comment (agarwal-navin) [inline packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:277]: `summaryRefSeq` should be passed directly here. There is no need to extract the `referenceSequenceNumber` from the snapshot.

**Review by agarwal-navin — APPROVED** [review]

**Pull request merged**